### PR TITLE
feat: implement delete function for k8s objects

### PR DESCRIPTION
**Pull request by tinkerborg**
This implements `delete` functionality for the standard-k8s trigger.

resolves argoproj/argo-events#1040
fixes argoproj/argo-events#638

### CODE REVIEW

**Comment by whynowy** [comment]
@VaibhavPage - Is there any reason we didn't do `DELETE` before?

File: sensors/triggers/standard-k8s/standar-k8s.go [hunk 1]
```diff
@@ -212,6 +212,22 @@ func (k8sTrigger *StandardK8sTrigger) Execute(ctx context.Context, events map[st
 
 		return k8sTrigger.namespableDynamicClient.Namespace(namespace).Patch(ctx, obj.GetName(), k8sTrigger.Trigger.Template.K8s.PatchStrategy, body, metav1.PatchOptions{})
 
+	case v1alpha1.Delete:
+		k8sTrigger.Logger.Info("deleting the object...")
+		_, err := k8sTrigger.namespableDynamicClient.Namespace(namespace).Get(ctx, obj.GetName(), metav1.GetOptions{})
+
+		if err != nil && apierrors.IsNotFound(err) {
+			k8sTrigger.Logger.Info("object not found, nothing to delete...")
```
Comment (whynowy) [inline sensors/triggers/standard-k8s/standar-k8s.go:220]: Need to return something here.

**Review by whynowy — CHANGES_REQUESTED** [review]

File: sensors/triggers/standard-k8s/standar-k8s.go (hunk 1 above)
Comment (tinkerborg) [reply sensors/triggers/standard-k8s/standar-k8s.go:220]: I made it return nil, nil for no-op, which I'm pretty sure makes sense here

**Comment by whynowy** [comment]
Fix the DCO please.

**Review by whynowy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**